tools/custom_builder/main.go

```diff
@@ -0,0 +1,153 @@
+
+or use one or more remote address(es) to load the config
+
+  custom_builder --config global.conf --config http://myserver/plugins.conf
+
+Combinations of local and remote config as well as config directories are
+possible.
+`
+
+func usage() {
+	_, _ = fmt.Fprint(flag.CommandLine.Output(), description)
+	_, _ = fmt.Fprintln(flag.CommandLine.Output(), "")
+	_, _ = fmt.Fprintln(flag.CommandLine.Output(), "Usage:")
+	_, _ = fmt.Fprintln(flag.CommandLine.Output(), "  custom_builder [flags]")
+	_, _ = fmt.Fprintln(flag.CommandLine.Output(), "")
+	_, _ = fmt.Fprintln(flag.CommandLine.Output(), "Flags:")
+	flag.PrintDefaults()
+	_, _ = fmt.Fprintln(flag.CommandLine.Output(), "")
+	_, _ = fmt.Fprintln(flag.CommandLine.Output(), "Examples:")
+	_, _ = fmt.Fprint(flag.CommandLine.Output(), examples)
+	_, _ = fmt.Fprintln(flag.CommandLine.Output(), "")
+}
+
+func main() {
+	var dryrun, showtags, quiet bool
+	var configFiles, configDirs []string
+
+	flag.Func("config",
+		"Import plugins from configuration file (can be used multiple times)",
+		func(s string) error {
+			configFiles = append(configFiles, s)
+			return nil
+		},
+	)
+	flag.Func("config-dir",
+		"Import plugins from configs in the given directory (can be used multiple times)",
+		func(s string) error {
+			configDirs = append(configDirs, s)
+			return nil
+		},
+	)
+	flag.BoolVar(&dryrun, "dry-run", false, "Skip the actual building step")
+	flag.BoolVar(&quiet, "quiet", false, "Print fewer log messages")
+	flag.BoolVar(&showtags, "tags", false, "Show build-tags used")
+
+	flag.Usage = usage
+	flag.Parse()
+
+	// Check configuration options
+	if len(configFiles) == 0 && len(configDirs) == 0 {
+		log.Fatalln("No configuration specified!")
+	}
+
+	// Import the plugin list from Telegraf configuration files
+	log.Println("Importing configuration file(s)...")
+	cfg, nfiles, err := ImportConfigurations(configFiles, configDirs)
+	if err != nil {
+		log.Fatalf("Importing configuration(s) failed: %v", err)
+	}
+	if !quiet {
+		log.Printf("Found %d configuration files...", nfiles)
+	}
+
+	// Check if we do have a config
+	if nfiles == 0 {
+		log.Fatalln("No configuration files loaded!")
+	}
+
+	// Collect all available plugins
+	packages := packageCollection{}
+	if err := packages.CollectAvailable(); err != nil {
+		log.Fatalf("Collecting plugins failed: %v", err)
+	}
+
+	// Process the plugin list with the given config. This will
+	// only keep the plugins that adhere to the filtering criteria.
+	enabled, err := cfg.Filter(packages)
+	if err != nil {
+		log.Fatalf("Filtering plugins failed: %v", err)
+	}
+	if !quiet {
+		enabled.Print()
+	}
+
+	// Extract the build-tags
+	tagset := enabled.ExtractTags()
+	if len(tagset) == 0 {
+		log.Fatalln("Nothing selected!")
+	}
+	tags := "custom," + strings.Join(tagset, ",")
+	if showtags {
+		fmt.Printf("Build tags: %s\n", tags)
+	}
+
+	if !dryrun {
+		// Perform the build
+		var out bytes.Buffer
+		makeCmd := exec.Command("make", "clean", "all")
```

Comment:
This seems to work, but it is a little scary we are removing the binary we are running and then building it again as a part of this whole process.